tfhe/src/shortint/ciphertext/compressed_ciphertext.rs

```diff
@@ -0,0 +1,171 @@
+                    );
+
+                    PackedModulusSwitchedLweCiphertext::compress(
+                        &ciphertext_buffers.buffer_lwe_after_ks.as_view(),
+                        CiphertextModulusLog(self.bootstrapping_key.polynomial_size().log2().0 + 1),
+                    )
+                });
+
+                PackedModulusSwitchedCiphertext(ms_ed)
+            }
+            PBSOrder::BootstrapKeyswitch => {
+                PackedModulusSwitchedCiphertext(PackedModulusSwitchedLweCiphertext::compress(
+                    &ct.ct.as_view(),
+                    CiphertextModulusLog(self.bootstrapping_key.polynomial_size().log2().0 + 1),
+                ))
+            }
+        }
+    }
+    // Unpacks a packed ciphertext
+    // This operation uses a PBS wo we can apply a lookup table
+    // An identity lookup table may be applied to get the pre packing ciphertext with a nominal
+    // noise
+    pub fn unpack_and_apply_lookup_table(
+        &self,
+        packed_ct: &PackedModulusSwitchedCiphertext,
+        acc: &LookupTableOwned,
+    ) -> Ciphertext {
+        let ct = packed_ct.0.extract();
+
+        let mut output = LweCiphertext::from_container(
+            vec![0; self.ciphertext_lwe_dimension().to_lwe_size().0],
+            self.ciphertext_modulus,
+        );
+
+        ShortintEngine::with_thread_local_mut(|engine| {
+            let (mut ciphertext_buffers, buffers) = engine.get_buffers(self);
+            match self.pbs_order {
+                PBSOrder::KeyswitchBootstrap => {
+                    apply_bootstrap(
+                        &self.bootstrapping_key,
+                        &ct.as_view(),
+                        &mut output.as_mut_view(),
+                        acc,
+                        buffers,
+                    );
+                }
+                PBSOrder::BootstrapKeyswitch => {
+                    apply_bootstrap(
+                        &self.bootstrapping_key,
+                        &ct.as_view(),
+                        &mut ciphertext_buffers.buffer_lwe_after_pbs,
+                        acc,
+                        buffers,
+                    );
+
+                    keyswitch_lwe_ciphertext(
+                        &self.key_switching_key,
+                        &ciphertext_buffers.buffer_lwe_after_pbs,
+                        &mut output.as_mut_view(),
+                    );
+                }
+            }
+        });
+
+        Ciphertext::new(
+            output,
+            acc.degree,
+            NoiseLevel::NOMINAL,
+            self.message_modulus,
+            self.carry_modulus,
+            self.pbs_order,
+        )
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use rand::Rng;
+
+    use crate::shortint::keycache::KEY_CACHE;
+    use crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    use crate::shortint::PBSParameters;
+
+    const NB_TESTS: usize = 10;
+
+    // create_parametrized_test!(shortint_message_extract);
```

Comment:
no parameterized test ?

Reply:
It's because the macro is not defined here
Not sure where I should put this test
I could move it with other shortint tests, but the file is really big
Or I could keep it here and copy the macro here, which isn't great

Reply:
server_key/test.rs ?

Reply:
> server_key/test.rs ?

What do you mean ?
Do you suggest moving this test to `server_key/tests/shortint.rs`?
This file is 3400 lines long, maybe we should split it

Reply:
Fixed all macro scoping